docs/testrunner-toolkit/configuration/xcuitest.md

```diff
@@ -1,190 +1,468 @@
+A set of parameters allowing you to provide additional details about which test class should be run for the suite and how to apply them.
 
-__Type__: *array*
+```yaml
+testOptions:
+    class:
+      - SwagLabsMobileAppUITests.LoginTests/testSuccessfulLogin
+      - SwagLabsMobileAppUITests.LoginTests/testNoUsernameLogin
+      - SwagLabsMobileAppUITests.LoginTests
+```
+---
+
+#### `class`
+<p><small>| OPTIONAL | ARRAY |</small></p>
+
+Instructs `saucectl` to only run the specified classes for this test suite.
 
-__Example__:
 ```yaml
-  class:
-    - SwagLabsMobileAppUITests.LoginTests/testSuccessfulLogin
-    - SwagLabsMobileAppUITests.LoginTests/testNoUsernameLogin
-    - SwagLabsMobileAppUITests.LoginTests
-```
\ No newline at end of file
+    class:
+      - SwagLabsMobileAppUITests.LoginTests/testSuccessfulLogin
+      - SwagLabsMobileAppUITests.LoginTests/testNoUsernameLogin
+      - SwagLabsMobileAppUITests.LoginTests
+```
+---
+
+## Advanced Configuration Considerations
+
+The configuration file is flexible enough to allow for any customizations and definitions that are required for any of the supported frameworks. The following sections describe some of the most common configurations.
+
+### Setting up a Proxy
+
+If you need to go through a proxy server, you can set it through the following variables:
+
+* `HTTP_PROXY`: Proxy to use to access HTTP websites
+* `HTTPS_PROXY`: Proxy to use to access HTTPS websites
+
+``` title= "Example: Windows Powershell"
+PS> $Env:HTTP_PROXY=http://my.proxy.org:3128/
+PS> $Env:HTTPS_PROXY=http://my.proxy.org:3128/
+```
+
+``` title= "Example: Linux/MacOS"
+$> export HTTP_PROXY=http://my.proxy.org:3128/
+$> export HTTPS_PROXY=http://my.proxy.org:3128/
+```
+
+
+### Tailoring Your Test File Bundle
+
+The `saucectl` command line bundles your root directory (`rootDir` parameter of `config.yml`) and transmits it to the Sauce Labs cloud, then unpacks the bundle and runs the tests. This functionality is partly what allows Sauce Control to operate in a framework-agnostic capacity. However, you can and should manage the inclusion and exclusion of files that get bundled to optimize performance and ensure security.
+
+#### Excluding Files from the Bundle
+
+The `.sauceignore` file allows you to designate certain files to be excluded from bundling.
+
+Add any files that are not direct test dependencies to `.sauceignore` to reduce the size of your bundle, improve test speed, and protect sensitive information.
+
+Examples of what can be included in `.sauceignore`:
+
+```bash
+# .sauceignore
+
+# Ignore node_modules
+node_modules/
+
+# Ignore all log files
+*.log
+
+# Ignore executables/binaries
+*.exe
+*.bin
+**/*/bin
+
+# Ignore media files
+*.png
+*.jpeg
+*.jpg
+*.mp4
+
+# Ignore documentation
+*.rst
+*.md
+
+# Ignore sensitive data
+credentials.yml
+```
```

Comment:
When running XCUITest tests, saucectl uploads only the `xcuitest.app` and `xcuitest.testApp` apps.
Even if creating a `.sauceignore` is possible, it will have no impact at all (its effect is only when packing test files for web-testing).

```suggestion
```